lnwallet/channel.go

```diff
@@ -6993,6 +6993,30 @@ func NewLocalForceCloseSummary(chanState *channeldb.OpenChannel,
 	}, nil
 }
 
+// chanCloseOpt is a functional option that can be used to modify the co-op
+// close process.
+type chanCloseOpt struct {
+	musigSession *MusigSession
+}
+
+// ChanCloseOpt is a closure type that cen be used to modify the set of default
+// options.
+type ChanCloseOpt func(*chanCloseOpt)
+
+// defaultCloseOpts is the default set of close options.
+func defaultCloseOpts() *chanCloseOpt {
+	return &chanCloseOpt{}
+}
+
+// WithCoopCloseMusigSession can be used to apply an existing musig2 session to
+// the cooperative close process. If specified, then a musig2 co-op close
+// (single sig keyspend) will be used.
+func WithCoopCloseMusigSession(session *MusigSession) ChanCloseOpt {
```

Comment:
What is the reason for choosing opts? Is it to keep the changeset manageable and not update every call with a new param?

I am thinking of a future time when taproot channels are default and v0 channels no longer supported. Then these opts will remain even though they are always set.

Reply:
Was mainly a way to extend the method without breaking the type signature. 

> I am thinking of a future time when taproot channels are default and v0 channels no longer supported. Then these opts will remain even though they are always set.

Based on the fact that even today some implementations still use the non-tweakless channels, I'd think that's a few years off....

If/when that comes to pass though, then we can remove these options.